language/stdlib/modules/Diem.move

```diff
@@ -540,13 +601,15 @@ module Diem {
     /// `CoinType` is empty.
     public fun burn_with_capability<CoinType>(
         preburn_address: address,
+        amount: u64,
         capability: &BurnCapability<CoinType>
     ) acquires CurrencyInfo, Preburn {
         // destroy the coin in the preburn to_burn area
         assert(exists<Preburn<CoinType>>(preburn_address), Errors::not_published(EPREBURN));
         burn_with_resource_cap(
```

Comment:
might need to stay the same
borrow the Conccuret, grabt he right now, mutable preburn and stick it in here (move_to takes from storgae to stack) so after the burn with resource cap call, 
let preburn = cuncurrent_preburns.get(...);
burn_with_resource_cap( ..., &mut preburn, ...);
let Preburn<Coin>{ coin } = preburn;
destroy_zero(coin);